app/services/oregon_digital/institution_picker.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+module OregonDigital
+  # A class that returns acronym or full name of two main institutions. Returns empty string for anything else
+  class InstitutionPicker
+    def self.institution_acronym(collection)
+      return '' unless contains_university?(collection)
+
+      institution = selected_institution(collection)
+      translate_institution(institution, :acronym)
+    end
+
+    def self.institution_full_name(collection)
+      return '' unless contains_university?(collection)
+
+      institution = selected_institution(collection)
+      translate_institution(institution, :full)
+    end
+
+    def self.selected_institution(collection)
+      institution = collection.institution.select { |val| val == osu }.first
+      institution = collection.institution.select { |val| val == uo }.first if institution.blank?
+      institution
+    end
+
+    def self.translate_institution(institution, translation_type)
+      case translation_type
+      when :acronym
+        institution == osu ? 'OSU' : 'UO'
+      when :full
+        institution == osu ? 'Oregon State University' : 'University of Oregon'
+      else
+        ''
+      end
+    end
+
+    def self.contains_university?(collection)
+      collection.institution.include?(uo) || collection.institution.include?(osu)
```

Comment:
Maybe we can do this for easier extension later?
`collection.institution.intersection(uo + osu).empty?`

My only concern is if there ends up being more URIs that get used to point to OSU/UO. I've seen one for dbpedia or something similar.

Reply:
This obviously makes a lot of other logic in here need some changes to revolve around arrays